pkg/controller/dns/provider.go

```diff
@@ -0,0 +1,86 @@
+package dns
+
+import (
+	"strings"
+	"time"
+
+	"github.com/acorn-io/acorn/pkg/dns"
+	"github.com/go-acme/lego/v4/challenge/dns01"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	AuthorizationHeader = "Authorization"
+	ContentTypeHeader   = "Content-Type"
+	ContentTypeJSON     = "application/json"
+	txtPathPattern      = "%s/domain/%s/txt"
+)
+
+/*
+ * DNS01 Challenge Solver (Lego Interface)
+ */
+
+type DNSProvider struct {
+	client DNSClient
+}
+
+func NewDNSProvider(endpoint, domain, token string) *DNSProvider {
```

Comment:
For clarity, can you rename this `NewACMEDNS01ChallengeProvider`. It wasn't clear to me what it was for at first

Reply:
Done.